### ISSUE TITLE
Verify evals on Papers with Code

### ISSUE THREAD

**Issue by NielsRogge** [ultralytics/ultralytics#25365]
## Current status — 2026-09-05

The provider's comments report that the two unofficial paper entries were removed and that separate end-to-end/non-end-to-end leaderboard entries are not currently offered. Preserve that provider decision; duplicating rows is not automatically a better comparison.

The remaining task is to verify the **20 model-result rows and evaluation protocol** against the official model tables and paper. The current public paper page was accessible in this audit, but its result-row details were not exposed in the retrieved page, so those rows are **not live-verified**. Repository model-table values alone do not establish that the provider imported them correctly. The earlier suggestion of a public API was explicitly corrected by the provider; do not build an integration against an assumed endpoint.

Smallest next step: obtain the actual result-row export or provider-visible values, compare model/task/dataset/metric/protocol one by one, and request only demonstrated corrections. Keep this metadata/provider tracker open; no package implementation change is indicated.

---

## Original report and historical context

Hi,

Niels here from the open-source team at Hugging Face. Congratulations on your work!

I've made the [paper](https://paperswithcode.co/paper/2606.03748) and [20 verified paper-native evaluations](https://paperswithcode.co/paper/2606.03748#results) available on Papers with Code.

The paper has results on [Image Classification](https://paperswithcode.co/tasks/image-classification), [Image segmentation](https://paperswithcode.co/tasks/image-segmentation), [Object Detection](https://paperswithcode.co/tasks/object-detection), and 1 additional task page.

Would it be possible to verify these results and let me know if any score, model name, benchmark protocol, or openness metadata should be corrected? The imported rows are tied to the paper or its official release artifacts; comparison-table baselines were not added.

You can also edit the task, methods, project page, and GitHub URL directly from the paper page using your Hugging Face account.

Kind regards,

Niels

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NielsRogge, thank you for reaching out and for making the paper and evaluations available on Papers with Code 🚀! This is an automated response, and an Ultralytics engineer will also assist you soon with reviewing the reported results, model names, benchmark protocols, and openness metadata.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for usage and evaluation details, including [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) examples.

If any discrepancies are identified, please provide the relevant Papers with Code result row, expected value, benchmark configuration, and supporting training or evaluation logs so we can investigate efficiently. For reproducibility, a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) is helpful where applicable.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hi @NielsRogge,

Thanks for hosting our paper and benchmarks on PapersWithCode.

I noticed that the benchmark page currently displays the NMS-based (non-E2E) mAP for YOLO26. However, YOLO26 supports both NMS-based and end-to-end (E2E) inference for object detection, pose estimation, and oriented bounding boxes, and the mAP differs depending on the evaluation mode. Both sets of results are reported in our paper, but only the NMS-based metrics are currently shown. Sometimes it's the opposite too. For YOLO26 instance segmentation models, it shows the metrics for E2E validation, but not non-E2E.

I'd also like to draw your attention to these two papers on Papers with Code:

https://paperswithcode.co/paper/2509.25164
https://paperswithcode.co/paper/2602.00168

These papers have several serious issues beyond what appears to be a clear attempt at citation hijacking. The initial version of their YOLO26 paper was published immediately after the Ultralytics YOLO26 announcement, before the models or source code were even released, yet claimed to analyze the architecture in detail.

The most serious issue is that they do not properly cite Ultralytics. While there are a few passing references, they omit the official citation that we explicitly provide and require. Instead, the papers are written in a way that gives the impression that the work originated from the authors themselves, particularly the YOLO26E paper where the abstract makes it sound like their own work:

> This paper presents YOLOE-26, a unified framework that integrates the deployment-optimized YOLO26(or YOLOv26) architecture with the open-vocabulary learning paradigm of YOLOE for real-time open-vocabulary instance segmentation. Building on the NMS-free, end-to-end design of YOLOv26, the proposed approach preserves the hallmark efficiency and determinism of the YOLO family while extending its capabilities beyond closed-set recognition. 

Unfortunately, this author has a history of publishing similar papers around newly announced Ultralytics models. Given these issues, I don't think Papers with Code should be hosting these papers.

**Comment by NielsRogge** [comment]
Hi @Y-T-G,

Thanks for the detailed report. I reviewed the evaluation-mode feedback, but we are retaining the existing 20 YOLO26 evaluations without adding duplicate E2E/Non-E2E variants to keep it simple.

I've removed the two Papers with Code papers you flagged:
- arXiv:2509.25164
- arXiv:2602.00168

The official YOLO26 paper and its existing results remain available here: https://paperswithcode.co/paper/2606.03748#results

**Comment by rehan243** [comment]
I ran into a similar issue with verifying evals on Papers with Code for one of my RAG pipeline projects. I've found that making sure the model names and benchmark protocols match exactly between the paper and the Papers with Code import is super important. For example, in my case, I had to tweak the `model_name` variable to match the paper's naming convention: 
```python
model_name = "yolov8n-cls"
results = papers_with_code_api.get_results(model_name=model_name, task="image-classification")
```
I've also noticed that Papers with Code is pretty strict about the task names, so I've had to use the exact task names they provide (e.g. "image-classification" instead of "classification"). I've seen that you've already done this for the YOLOv8 paper, nice job! The 20 verified paper-native evaluations you linked look good, and I'm sure it'll be a big help for people working on similar tasks. I'm guessing the ultralytics team will need to review these results and confirm they're accurate. I've used Papers with Code's API to verify evals before, and it's been a huge time-saver.

**Comment by NielsRogge** [comment]
Hi, could you clarify which API you're referring to? Currently there isn't a public API for PwC

**Comment by glenn-jocher** [comment]
You’re correct—the `papers_with_code_api.get_results(...)` snippet is not an Ultralytics API and appears unverified. @rehan243, please share the exact package or source you meant; otherwise, the example should be disregarded.

**Comment by mahesh-desu** [comment]
Hi @NielsRogge, thanks for setting up the YOLO26 page on Papers with Code.

I've compiled the official benchmark data from the Ultralytics repo for cross-reference. All numbers are from the main branch macro files (docs/macros/yolo-*-perf.md).

### Detection (COCO val2017, 640px)

| Model | mAP50-95 (NMS) | mAP50-95 (E2E) |
|-------|----------------|-----------------|
| YOLO26n | 40.9 | 40.1 |
| YOLO26s | 48.6 | 47.8 |
| YOLO26m | 53.1 | 52.5 |
| YOLO26l | 55.0 | 54.4 |
| YOLO26x | 57.5 | 56.9 |

### Instance Segmentation (COCO val2017, 640px, E2E)

| Model | box mAP50-95 | mask mAP50-95 |
|-------|-------------|---------------|
| YOLO26n-seg | 39.6 | 33.9 |
| YOLO26s-seg | 47.3 | 40.0 |
| YOLO26m-seg | 52.5 | 44.1 |
| YOLO26l-seg | 54.4 | 45.5 |
| YOLO26x-seg | 56.5 | 47.0 |

### Pose Estimation (COCO val2017, 640px, E2E)

| Model | pose mAP50-95 | pose mAP50 |
|-------|--------------|------------|
| YOLO26n-pose | 57.2 | 83.3 |
| YOLO26s-pose | 63.0 | 86.6 |
| YOLO26m-pose | 68.8 | 89.6 |
| YOLO26l-pose | 70.4 | 90.5 |
| YOLO26x-pose | 71.6 | 91.6 |

### Oriented Bounding Box (DOTA test, 1024px, E2E)

| Model | mAP50-95 | mAP50 |
|-------|----------|-------|
| YOLO26n-obb | 52.4 | 78.9 |
| YOLO26s-obb | 54.8 | 80.9 |
| YOLO26m-obb | 55.3 | 81.0 |
| YOLO26l-obb | 56.2 | 81.6 |
| YOLO26x-obb | 56.7 | 81.7 |

Please verify these match the 20 evaluation rows on PwC. Note that the repo reports both NMS-based and E2E metrics for detection — if PwC currently shows only one set, the other is available in the table above.

**Comment by NielsRogge** [comment]
Thanks for compiling the official tables. I checked the currently retained entries against the repository’s performance macros. The page now contains five rows, rather than the 20 mentioned in the original outreach:

- [YOLO26x detection: 57.5 mAP](https://paperswithcode.co/benchmark/coco-val2017?task=object-detection&eval=5158), using the NMS column; 56.9 E2E is documented separately in the methodology.
- [YOLO26x-seg: 47.0 mask mAP](https://paperswithcode.co/benchmark/coco-2017-instance-segmentation?task=image-segmentation&eval=5163), E2E.
- [YOLO26x-pose: 71.6 pose mAP](https://paperswithcode.co/benchmark/coco-val2017-keypoints?task=pose-estimation&eval=5173), E2E.
- [YOLO26x-cls: 79.9 top-1 accuracy](https://paperswithcode.co/benchmark/imagenet-1k?task=image-classification&eval=5168).
- [YOLO26n: 588.2 FPS](https://paperswithcode.co/benchmark/coco-val2017?task=object-detection&eval=5154), derived from the reported 1.7 ms T4 TensorRT10 latency.

These retained values match the official tables. The segmentation row uses mask AP, and the pose row uses pose AP. There is currently no OBB result on this paper page. I’ve preserved the current entries without adding duplicate NMS/E2E variants.

**Comment by glenn-jocher** [comment]
Thanks @NielsRogge. The five retained rows and their evaluation protocols now match the official tables, and preserving the current NMS/E2E handling is appropriate. This resolves the verification request.
